lucille-core/src/test/java/com/kmwllc/lucille/stage/ConfigValidationTest.java

```diff
@@ -57,33 +58,80 @@ public void testNonValidationModeException() throws Exception {
     assertFalse(result.getStatus());
   }
 
+  // asserts that if two connectors use the same pipeline it is only validated once
   @Test
-  public void testValidationModeException() throws Exception {
-    Map<String, List<Exception>> exceptions = Runner.runInValidationMode(addPath("pipeline.conf"));
-    assertEquals(2, exceptions.size());
+  public void testDuplicatePipeline() throws Exception {
+    Map<String, List<Exception>> exceptions = Runner.runInValidationMode(addPath("duplicate-pipeline.conf"));
+    assertEquals(1, exceptions.size());
 
-    List<Exception> exceptions1 = exceptions.get("connector1");
+    List<Exception> exceptions1 = exceptions.get("pipeline1");
     assertEquals(2, exceptions1.size());
 
-    List<Exception> exceptions2 = exceptions.get("connector2");
-    assertEquals(2, exceptions2.size());
+    testException(exceptions1.get(0), IllegalArgumentException.class,
+        "com.kmwllc.lucille.stage.NoopStage: " + "Stage config contains unknown property invalid_property");
+    testException(exceptions1.get(1), IllegalArgumentException.class, "Stage config must contain property fields");
+  }
 
-    testException(exceptions1.get(0), IllegalArgumentException.class, "com.kmwllc.lucille.stage.NoopStage: " +
-        "Stage config contains unknown property invalid_property");
+  // asserts that if no connectors use a pipeline it is still validated
+  @Test
+  public void testUnusedPipeline() throws Exception {
+    Map<String, List<Exception>> exceptions = Runner.runInValidationMode(addPath("no-used-pipelines.conf"));
+    assertEquals(1, exceptions.size());
+
+    List<Exception> exceptions1 = exceptions.get("pipeline1");
+    assertEquals(2, exceptions1.size());
 
-    // TODO note that for the following two exceptions, the fields are retrieved before
-    //  the config validation is called
+    testException(exceptions1.get(0), IllegalArgumentException.class,
+        "com.kmwllc.lucille.stage.NoopStage: " + "Stage config contains unknown property invalid_property");
```

Comment:
this is on one line so we don't need the plus. same issue occurs in one or two other places